File: app/src/main/java/medlife/com/br/helper/CartManager.java
package medlife.com.br.helper;

import java.util.ArrayList;
import java.util.List;
import medlife.com.br.model.CartItem;
import medlife.com.br.model.Product;
import medlife.com.br.model.Order;
import medlife.com.br.model.OrderStatus;
import java.util.HashMap;
import java.util.Map;

public class CartManager {
    private static CartManager instance;
    private List<CartItem> cartItems = new ArrayList<>();

    private CartManager() {}

    public static synchronized CartManager getInstance() {
        if (instance == null) {
            instance = new CartManager();
        }
        return instance;
    }

    public void addProduct(Product product, int quantity) {
        for (CartItem item : cartItems) {
            if (item.getProduct().getName().equals(product.getName())) {
                item.setQuantity(item.getQuantity() + quantity);
                return;
            }
        }
        cartItems.add(new CartItem(product, quantity));
    }

    public void removeProduct(CartItem cartItem) {
        cartItems.remove(cartItem);
    }

    public void updateQuantity(CartItem cartItem, int newQuantity) {
        if (newQuantity > 0) {
            cartItem.setQuantity(newQuantity);
        } else {
            removeProduct(cartItem);
        }
    }

    public double getTotalPrice() {
        double total = 0;
        for (CartItem item : cartItems) {
            String priceString = item.getProduct().getPrice().replace("R$", "").replace(",", ".").trim();
            try {
                total += Double.parseDouble(priceString) * item.getQuantity();
            } catch (NumberFormatException e) {
                // Ignore products with invalid price formats
            }
        }
        return total;
    }

    public void clearCart() {
        cartItems.clear();
    }

    public List<CartItem> getCartItems() {
        return cartItems;
    }

    public Order createOrderFromCart() {
        String userId = UsuarioFirebase.getIdUsuario();
        if (userId == null || cartItems.isEmpty()) {
            return null;
        }

        List<Map<String, Object>> orderItems = new ArrayList<>();
        for (CartItem item : cartItems) {
            Map<String, Object> orderItem = new HashMap<>();
            orderItem.put("productName", item.getProduct().getName());
            orderItem.put("quantity", item.getQuantity());
            orderItem.put("price", item.getProduct().getPrice());
            orderItem.put("totalItemPrice", Double.parseDouble(item.getProduct().getPrice().replace("R$", "").replace(",", ".").trim()) * item.getQuantity());
            orderItem.put("category", item.getProduct().getCategory());
            orderItem.put("brand", item.getProduct().getBrand());
            orderItem.put("tarja", item.getProduct().getTarja());
            orderItem.put("description", item.getProduct().getDescription());
            orderItems.add(orderItem);
        }

        Order order = new Order();
        order.setUserId(userId);
        order.setItems(orderItems);
        order.setTotalPrice(getTotalPrice());
        order.setStatus(OrderStatus.ORDER_CONFIRMED.toString());
        
        // Set subtotal
        order.setSubtotal(getTotalPrice());
        
        // Set default values for other fields
        // These will be updated by CartFragment before saving
        order.setDeliveryOption("immediate");
        order.setDeliveryFee(7.00);
        order.setDiscountAmount(0.0);
        order.setPaymentMethod("Pendente");
        order.setPaymentStatus("pending");
        order.setStoreName("Drogaria São Paulo");
        order.setStoreLocation("Cidade Campinas");
        order.setRequiresPrescription(false);

        return order;
    }
} 

File: app/src/main/java/medlife/com/br/helper/CartManager.kt
package medlife.com.br.helper

import medlife.com.br.model.CartItem
import medlife.com.br.model.Order
import medlife.com.br.model.OrderStatus
import medlife.com.br.model.Product

object CartManager {

    private val cartItems: MutableList<CartItem> = mutableListOf()

    fun addProduct(product: Product, quantity: Int) {
        val existing = cartItems.find { it.product.name == product.name }
        if (existing != null) {
            existing.quantity += quantity
        } else {
            cartItems.add(CartItem(product, quantity))
        }
    }

    fun removeProduct(cartItem: CartItem) {
        cartItems.remove(cartItem)
    }

    fun updateQuantity(cartItem: CartItem, newQuantity: Int) {
        if (newQuantity > 0) {
            cartItem.quantity = newQuantity
        } else {
            removeProduct(cartItem)
        }
    }

    fun getTotalPrice(): Double {
        return cartItems.sumOf { item ->
            val priceString = item.product.price
                .replace("R$", "")
                .replace(",", ".")
                .trim()
            try {
                priceString.toDouble() * item.quantity
            } catch (e: NumberFormatException) {
                0.0
            }
        }
    }

    fun clearCart() {
        cartItems.clear()
    }

    fun getCartItems(): List<CartItem> = cartItems

    fun createOrderFromCart(): Order? {
        val userId = UsuarioFirebase.getIdUsuario() ?: return null
        if (cartItems.isEmpty()) return null

        val orderItems: List<Map<String, Any>> = cartItems.map { item ->
            val price = item.product.price
                .replace("R$", "")
                .replace(",", ".")
                .trim()
                .toDoubleOrNull() ?: 0.0
            mapOf(
                "productName" to (item.product.name ?: ""),
                "quantity" to item.quantity,
                "price" to (item.product.price ?: ""),
                "totalItemPrice" to (price * item.quantity),
                "category" to (item.product.category ?: ""),
                "brand" to (item.product.brand ?: ""),
                "tarja" to (item.product.tarja ?: ""),
                "description" to (item.product.description ?: ""),
                "farmacia" to (item.product.farmacia ?: "")
            )
        }

        return Order().apply {
            this.userId = userId
            this.items = orderItems
            this.totalPrice = getTotalPrice()
            this.status = OrderStatus.ORDER_CONFIRMED.toString()
            this.subtotal = getTotalPrice()
            this.deliveryOption = "immediate"
            this.deliveryFee = 7.00
            this.discountAmount = 0.0
            this.paymentMethod = "Pendente"
            this.paymentStatus = "pending"
            this.requiresPrescription = false
        }
    }
}
